sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -143,6 +143,17 @@ object GpuDeviceManager extends Logging {
     singletonMemoryInitialized = Errored
     RapidsBufferCatalog.close()
     GpuShuffleEnv.shutdown()
+    // try to avoid segfault on RMM shutdown
+    val timeout = System.nanoTime() + TimeUnit.SECONDS.toNanos(10)
+    while(Rmm.getTotalBytesAllocated > 0 && System.nanoTime() < timeout) {
```

Comment:
nit: formatting
```suggestion
    while (Rmm.getTotalBytesAllocated > 0 && System.nanoTime() < timeout) {
```

I think we need a follow-up issue for to replace the sleep loop  with a proper condition await.